core/services/blockhashstore/feeder.go

```diff
@@ -114,3 +132,114 @@ func (f *Feeder) Run(ctx context.Context) error {
 	f.lastRunBlock = latestBlock
 	return errs
 }
+
+func (f *Feeder) runTrusted(
+	ctx context.Context,
+	latestBlock uint64,
+	fromBlock uint64,
+	blockToRequests map[uint64]map[string]struct{},
+) error {
+	var errs error
+
+	// Iterate through each request block via waitGroup.
+	// For blocks with pending requests, add them to the batch to be stored.
+	// Note: Golang maps sort items in a range randomly, so although the batch size is used
+	// to limit blocks-per-batch, every block has an equal chance of getting picked up
+	// on each run.
+	var batch = make(map[uint64]struct{})
+	for blockKey, unfulfilledReqs := range blockToRequests {
```

Comment:
Nit: can you do `for block, unfulfilled := ...` to avoid the `var block = blockKey` and `var unfulfilled = unfulfilledReqs` aliases below? Or is there some weird aliasing issue that comes with that.